os/fs/procfs/fs_procfslogsave.c

```diff
@@ -0,0 +1,191 @@
+/****************************************************************************
```

Comment:
you can add this codes to bottom of serial.c
please refer to mtd_partition.c
and something like a 'CONFIG_PROCFS_EXCLUDE_SAVEDLOG' should be there to exclude it if user doesn't want that function